x/auth/types/account.go

```diff
@@ -334,3 +404,121 @@ type GenesisAccount interface {
 
 	Validate() error
 }
+
+// custom json marshaler for BaseAccount & ModuleAccount
+
+type BaseAccountJSON struct {
+	Address  string          `json:"address"`
+	PubKey   json.RawMessage `json:"pub_key"`
+	Sequence string          `json:"sequence"`
+}
+
+func (acc BaseAccount) MarshalJSONPB(m *jsonpb.Marshaler) ([]byte, error) {
+	var bi BaseAccountJSON
+
+	bi.Address = acc.GetAddress().String()
+	bi.Sequence = strconv.FormatUint(acc.Sequence, 10)
+	var bz []byte
+	var err error
+	if acc.Ed25519PubKey != nil {
+		bz, err = codec.ProtoMarshalJSON(acc.Ed25519PubKey, m.AnyResolver)
+	}
+	if acc.Secp256K1PubKey != nil {
+		bz, err = codec.ProtoMarshalJSON(acc.Secp256K1PubKey, m.AnyResolver)
+	}
+	if acc.MultisigPubKey != nil {
+		bz, err = codec.ProtoMarshalJSON(acc.MultisigPubKey, m.AnyResolver)
+	}
+	if err != nil {
+		return nil, err
+	}
+	bi.PubKey = bz
+	return json.Marshal(bi)
+}
+
+func (acc *BaseAccount) UnmarshalJSONPB(m *jsonpb.Unmarshaler, bz []byte) error {
+	var bi BaseAccountJSON
+
+	err := json.Unmarshal(bz, &bi)
+	if err != nil {
+		return err
+	}
+	/* TODO: do we need to validate address format here
```

Comment:
Yes. I think we'd better validate the address here. I don't think this part affects performance much.

Reply:
The issue was we need to call three validators.

if err := sdk.ValidateAccAddress(bi.Address); err != nil {
    if err = sdk.ValidateValAddress(bi.Address); err != nil {
        if err = sdk.ValidateConsAdderss(bi.Address); err != nil {
            return err
        }
    }
}

Is there better way than this?

Reply:
Then let's do this. Only the address entered by the user is verified. It's probably msg handler.
And `Unmarshal` is not validate an address because it operates only on marshaled data.